Products/CMFPlone/relationhelper.py

```diff
@@ -148,101 +151,138 @@ def restore_relations(context=None, all_relations=None):
             unique_relations.append(rel)
             seen_add(hashable)
         else:
-            logger.info(f'Dropping duplicate: {hashable}')
+            logger.info(f"Dropping duplicate: {hashable}")
 
     if len(unique_relations) < len(all_relations):
-        logger.info(f'Dropping {len(all_relations) - len(unique_relations)} duplicates')
+        logger.info(f"Dropping {len(all_relations) - len(unique_relations)} duplicates")
         all_relations = unique_relations
 
     intids = getUtility(IIntIds)
     for index, item in enumerate(all_relations, start=1):
         if not index % 500:
-            logger.info(f'Restored {index} of {len(all_relations)} relations...')
+            logger.info(f"Restored {index} of {len(all_relations)} relations...")
 
+        # source object for UID exists.
         try:
-            source_obj = uuidToObject(item['from_uuid'])
+            source_obj = uuidToObject(item["from_uuid"])
         except KeyError:
             # brain exists but no object
             source_obj = None
+        finally:
+            if not source_obj:
+                logger.info(f'No source object found for UID {item["from_uuid"]}.')
+                continue
+
+        # target object for UID exists.
         try:
-            target_obj = uuidToObject(item['to_uuid'])
+            target_obj = uuidToObject(item["to_uuid"])
         except KeyError:
             # brain exists but no object
             target_obj = None
+        finally:
+            if not target_obj:
+                logger.info(f'No target object found for UID {item["to_uuid"]}.')
+                # Do not continue. The source_obj needs to be updated.
+                # continue
```

Comment:
```suggestion
                # The source_obj will be updated to remove the broken relation below.
```

(This is clearer to me. Using the word "continue" makes it hard to tell whether you mean continuing to the next iteration of the loop, as the `continue` statement does, or continuing to execute code below, which does happen now.)

Reply:
Yes, _continue_ could mean both in this context. Updated with your suggestion.